.github/workflows/build.yml

```diff
@@ -37,19 +37,61 @@ jobs:
           python setup.py bdist_wheel
 
       - name: Run Tests
-        run : pipenv run tests # execute Pipfile script tests
-
-      - name: Build Docs
-        run: pipenv run docs # build docs
+        run : pipenv run tests
 
       - name: Coveralls
-        # the official action only uses Lcov files which pytest-cov does not support
-        # https://github.com/coverallsapp/github-action/issues/30
-        #
-        # using https://github.com/coveralls-clients/coveralls-python instead
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           pip install coveralls
           coveralls
 
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout master
+        uses: actions/checkout@v2
+        with:
+          ref: master
+          # require all of history to build all tagged versions' docs
+          fetch-depth: 0
+
+      # also checkout the branch that triggered this build
+      - name: Checkout Source
+        uses: actions/checkout@v2
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Install Dependencies
+        run: |
+          pip install pipenv
+          pipenv install --dev
+
+      - name: Build Docs
+        # this builds all tagged versions and master
+        run: pipenv run sphinx-multiversion docs build/html
```

Comment:
Can we build directly in gh-pages rather than having to copy there?

Reply:
yes I think so - not sure why I did not do this :-)

Reply:
I tried this and got a strange issue (see revert). Not sure it is worth spending time on.